src/Components/People/EditPerson.tsx

```diff
@@ -0,0 +1,199 @@
+import {
+  CardContent,
+  CardActions,
+  TextField,
+  Typography,
+  Fab,
+  FormControl,
+  InputLabel,
+  Select,
+  MenuItem,
+  SelectChangeEvent,
+} from "@mui/material";
+import DoneIcon from "@mui/icons-material/Done";
+import ClearIcon from "@mui/icons-material/Clear";
+import SaveIcon from "@mui/icons-material/Save";
+import { useEffect, useState } from "react";
+import { personEnabledDetails } from "../../Interfaces/Person";
+import GroupServices from "../../Services/Groups";
+import { toast } from "react-toastify";
+import { IGroup } from "../../Interfaces/Group";
+
+const PersonEnabledDetails = (props: personEnabledDetails) => {
+  const {
+    setAreSlotsEnabled,
+    areSlotsEnabled,
+    updatePerson,
+    createPerson,
+    name,
+    favoriteFood,
+    favoriteAnimal,
+    favoriteColor,
+    id,
+  } = props;
+  const [newName, setName] = useState(name);
+  const [newFavoriteColor, setFavoriteColor] = useState(favoriteColor);
+  const [newFavoriteAnimal, setFavoriteAnimal] = useState(favoriteAnimal);
+  const [newFavoriteFood, setFavoriteFood] = useState(favoriteFood);
+  const [allGroups, setAllGroups] = useState();
+  const [group, setGroup] = useState("");
+
+  const handleChange = (event: SelectChangeEvent) => {
+    setGroup(event.target.value);
+  };
+
+  useEffect(() => {
+    GroupServices.getAllGroups()
+      .then((res) => {
+        setAllGroups(
+          res.data.map((group: IGroup) => {
+            return (
+              <MenuItem key={group.id} value={group.id}>
+                {group.name}
+              </MenuItem>
+            );
+          })
+        );
+      })
+      .catch((err) => {
+        return toast.error(err.response.data);
+      });
+  }, []);
+
+  return (
+    <CardContent>
+      <Typography variant="h5" component="div"></Typography>
+      <TextField
+        sx={{ mb: 0.4, mt: -0.5 }}
+        id="name-slot"
+        variant="standard"
+        placeholder="Name:"
+        value={newName}
+        onChange={(text) => setName(text.target.value)}
+      />
+      <TextField
+        InputProps={{ disableUnderline: true }}
+        sx={{ ml: 2, mb: -1 }}
+        id="details-slot"
+        variant="standard"
+        placeholder="Favorite color:"
+        value={newFavoriteColor}
+        onChange={(text) => setFavoriteColor(text.target.value)}
+      />
+      <TextField
+        InputProps={{ disableUnderline: true }}
+        sx={{ ml: 2, mb: -1 }}
+        id="details-slot"
+        variant="standard"
+        placeholder="Favorite food:"
+        value={newFavoriteFood}
+        onChange={(text) => setFavoriteFood(text.target.value)}
+      />
+      <TextField
+        InputProps={{ disableUnderline: true }}
+        sx={{ mb: -1, ml: 2 }}
+        id="details-slot"
+        placeholder="Favorite animal:"
+        variant="standard"
+        value={newFavoriteAnimal}
+        onChange={(text) => setFavoriteAnimal(text.target.value)}
+      />
```

Comment:
Lots of duplication here, could make a `map` on an array of fields like the following to generate it dynamically and make it much more readable.

```js
const inputFields = [
	{
		key: "name",
		placeholder: "Name: ",
	},
	{
		key: "favColor",
		placeholder: "Favorite Color: ",
	},
]
```

Also, you're reusing `id=details-slot`. CSS IDs are not meant to be reused.